pkg/network/network.go

```diff
@@ -178,6 +178,12 @@ func Render(conf *cnao.NetworkAddonsConfigSpec, manifestDir string, openshiftNet
 	}
 	objs = append(objs, o...)
 
+	o, err = RenderNetworkPolicy(manifestDir, *clusterInfo)
```

Comment:
what about removal please ?
unless thanks to relation labels (need to make sure anyhow they are updated please) it removes those once cnao is undeployed

can you please test different scenarios ?
i.e if HCO deploys it and removes CNAO, this should be removed as well
(there might be more scenarios)

if you find a way to simulate it or to see in the code / created objects that we are fine, it might be good enough

Reply:
It is done by [[1]](https://github.com/kubevirt/cluster-network-addons-operator/blob/main/pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go#L383)  for all rendered objects (except for the project  Namespace)